base/channels.jl

```diff
@@ -553,6 +665,132 @@ function take_unbuffered(c::Channel{T}) where T
     end
 end
 
+"""
+    take!(c::Channel, n::Integer[, buffer::AbstractVector])
+
+Take at most `n` items from the [`Channel`](@ref) `c` and return them in a `Vector`.
+If `buffer` is provided, it will be used to store the result. If the channel closes before
+`n` items are taken, the result will contain only the items that were available. For
+buffered channels, this operation requires fewer `lock` operations than individual `take!(c)`s.
+
+# Examples
+
+```jldoctest
+julia> c = Channel(3);
+
+julia> append!(c, 1:3);
+
+julia> take!(c, 3, zeros(Int, 3))
+3-element Vector{Int64}:
+ 1
+ 2
+ 3
+```
+
+!!! compat "Julia 1.12"
+    Requires at least Julia 1.12.
+"""
+function take!(c::Channel{T}, n::Integer) where {T}
+    return _take(c, n, Vector{T}(undef, n))
+end
+function take!(c::Channel, n::Integer, buffer::AbstractVector)
+    # buffer is user defined, so make sure it has the correct size
+    if length(buffer) != n
+        resize!(buffer, n)
+    end
```

Comment:
These lines prevent you from doing `take(channel, typemax(Int), buffer)` as an idiom to fill all elements of the channel into the buffer. It would be nice if that worked.

Reply:
Done!